stdnum/isl/vsk.py

```diff
@@ -0,0 +1,63 @@
+# vsk.py - functions for handling Icelandic VAT numbers
+# coding: utf-8
+#
+# Copyright (C) 2012, 2013 Arthur de Jong
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA
+
+"""VSK number (Virðisaukaskattsnúmer, Icelandic VAT number).
+
+The Icelandic VAT number is five or six digits.
+
+>>> validate('IS 00621')
+'00621'
+>>> validate('IS 0062199')  # invalid length
+Traceback (most recent call last):
+    ...
+InvalidChecksum: ...
+"""
+
+from stdnum.exceptions import *
+from stdnum.util import clean
+
+
+def compact(number):
+    """Convert the number to the minimal representation. This strips the
+    number of any valid separators and removes surrounding whitespace."""
+    number = clean(number, ' ').upper().strip()
+    if number.startswith('IS'):
+        number = number[2:]
+    return number
+
+
+def validate(number):
+    """Checks to see if the number provided is a valid VAT number. This
+    checks the length, formatting and check digit."""
+    number = compact(number)
+    if not number.isdigit():
+        raise InvalidFormat()
+    if len(number) != 5 and len(number) != 6:
+        raise InvalidLength()
+    return number
```

Comment:
@toivotuo doc string says this is supposed to check check digit as well, but i do not see that here, maybe it is only doc string which is not valid here?